lib/Service/DkimService.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2023 Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * @author Daniel Kesselberg <mail@danielkesselberg.de>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Mail\Service;
+
+use OCA\Mail\Account;
+use OCA\Mail\Contracts\IDkimService;
+use OCA\Mail\Contracts\IDkimValidator;
+use OCA\Mail\Db\Mailbox;
+use OCA\Mail\Exception\ServiceException;
+use OCA\Mail\IMAP\IMAPClientFactory;
+use OCA\Mail\IMAP\MessageMapper;
+use OCA\Mail\Model\DkimResult;
+use OCP\ICache;
+use OCP\ICacheFactory;
+use function json_encode;
+
+class DkimService implements IDkimService {
+	private const CACHE_PREFIX = 'mail_dkim_';
+	private const CACHE_TTL = 7 * 24 * 3600;
+
+	/** @var IMAPClientFactory */
+	private $clientFactory;
+
+	/** @var MessageMapper */
+	private $messageMapper;
+
+	/** @var ICache */
+	private $cache;
+
+	private IDkimValidator $dkimValidator;
+
+	public function __construct(
+		IMAPClientFactory $clientFactory,
+		MessageMapper $messageMapper,
+		ICacheFactory $cacheFactory,
+		IDkimValidator $dkimValidator
+	) {
+		$this->clientFactory = $clientFactory;
+		$this->messageMapper = $messageMapper;
+		$this->cache = $cacheFactory->createDistributed();
```

Comment:
```suggestion
		$this->cache = $cacheFactory->createDistributed(self::CACHE_PREFIX);
```

nit: to have the prefix added for you

Reply:
Good to know!